### ISSUE TITLE
bug(mat-dialog / mat-slide-toggle): mat-dialog-content always has a scroll bar when there is a mat-slide-toggle inside it

### ISSUE THREAD

**Issue by Ploppy3** [angular/components#20344]
#### Reproduction

Steps to reproduce:
put a mat-slide-toggle in a mat-dialog-content

https://stackblitz.com/edit/angular-pt8j22?file=src/app/dialog-content-example-dialog.html
 
#### Expected Behavior
there should not be a scroll bar when having a mat-slide-toggle in a mat-dialog-content if the content does not exceed the max height allowed

#### Actual Behavior
mat-dialog-content always has a scroll bar when there is a mat-slide-toggle inside it

#### Environment

- Angular: 10
- CDK/Material: 10
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by Totati** [comment]
Hi, this is caused by `.mat-slide-toggle .mat-slide-toggle-ripple` that has 40px height, which is larger than the `mat-slide-toggle`'s 24px. We overcome this problem by applying padding top/bottom to a the toggle, when it's first/last.

**Comment by Ploppy3** [comment]
@Totati Thank you for the workaround, I could not find the solution, much appreciated.

**Comment by nienow** [comment]
Just wanted to add that I ran into this issue with radio buttons as well, when they are at the bottom of the dialog content.

**Comment by Ploppy3** [comment]
@nienow Yes, it happens because radio buttons also have ripples.

**Comment by kyletanders** [comment]
Any update on this?  This has been an annoyance for a long time.  Would be nice to get it fixed.

**Comment by neerajPortalwiz** [comment]
Yeah having the same issue, with radio buttons at the bottom of dialog

**Comment by mpeguero** [comment]
Angular material 15: this solved the issue
  // to fix the issue with the dialog scroll issue
  .mdc-dialog__surface {
     overflow-y: hidden !important;
    }

**Comment by roma2341** [comment]
Its weird and unusable behavior

**Comment by carl-sauter** [comment]
So nearly 6 years now, and nothing has changed :(
